maven-model/src/main/java/org/apache/maven/model/merge/ModelMerger.java

```diff
@@ -3018,4 +2618,298 @@ protected Object getExclusionKey( Exclusion exclusion )
         return exclusion;
     }
 
+    interface KeyComputer<T>
```

Comment:
private?

Reply:
Yes, all new classes along with the interface could be made private.

Reply:
Maybe protected would be better, as they could be reused in derived classes.

Reply:
Definitely make it private. Java developer must be conservative.
If somebody has a reason to break the information hiding then he makes it protected or even more open.